src/Features/CSharp/Portable/Completion/CompletionProviders/DeclarationNameCompletionProvider.DeclarationInfo.cs

```diff
@@ -154,10 +154,18 @@ internal static async Task<NameDeclarationInfo> GetDeclarationInfoAsync(Document
                 return false;
             }
 
-            private static bool IsPossibleVariableOrLocalMethodDeclaration(
+            private static bool IsPossibleVariableOrLocalFunctionDeclaration(
                 SyntaxToken token, SemanticModel semanticModel,
                 CancellationToken cancellationToken, out NameDeclarationInfo result)
             {
+                // It is impossible to declare a variable or local function after an expression like:
+                // this.SomeMember $$
+                if (token.GetAncestor<ExpressionStatementSyntax>() is { Expression: MemberAccessExpressionSyntax { Expression: ThisExpressionSyntax } })
+                {
+                    result = default;
+                    return false;
+                }
```

Comment:
ok.  so i wouldn't go about it this wya.  instead, have IsLastTokenOfType return the ExprStatement as well.  then see if the ExprStatement.Expr is a TypeSyntax or not.  If not, it's not a possible variable decl.

Reply:
This will not work for cases like `System.Action` because it is a `SimpleMemberAccessExpression` that does not inherit from `TypeSyntax`

Reply:
Sure. You can also validate that every part of the expr is a name syntax. We probably shouldn't be using GetTypeInfo here either, but rather GetZymbolInfo and validating that that binds to an INamedType.